src/softwareHeritage/agent/softwareHeritageAgent.php

```diff
@@ -129,6 +129,7 @@ function processEachPfileForSWH($pfileDetail, $agentId, $maxTime)
       } else {
         sleep($timeToReset);
       }
+      print "INFO :Software Heritage X-RateLimit-Limit reached. Next slot unlocks in ".gmdate("H:i:s", $timeToReset)." ";
```

Comment:
IMO, the message should be printed before going to sleep.

Reply:
Does it really make a difference ??

Reply:
The idea of the message is to inform user why agent is not processing more files.

Also, let's say if the rate limit will reset in 30 seconds and you've already paused for 30 seconds and then inform user you are going to pause for 30 seconds, the math does not add up.